### ISSUE TITLE
Rosetta: Accept compressed public key for /derive /combine

### ISSUE THREAD

**Issue by bkase** [MinaProtocol/mina#9929]
This entails reading a public key encoded in the following manner (note: hex-encoded as usual like the uncompressed public keys)

```
func (p *Ep) ToAffineCompressed() []byte {
	// Use ZCash encoding where infinity is all zeros
	// and the top bit represents the sign of y and the
	// remainder represent the x-coordinate
	var inf [32]byte
	p1 := new(Ep).Set(p)
	p1.toAffine()
	x := p1.x.Bytes()
	x[31] |= (p1.y.Bytes()[0] & 1) << 7
	subtle.ConstantTimeCopy(bool2int[p1.IsIdentity()], x[:], inf[:])
	return x[:]
}
```

My understanding is that this should break down into the following changes:
- [ ] Check the length of the input; if it is 32 bytes use this new decode path. Otherwise keep the old 64 bytes path
- [ ] Decode the first bit as the "is_odd" boolean in our Public_key.Compressed format, and take the rest of the bits as the x coordinate
- [ ] Write a unit test for both 1 and 0 y-parity bits

**Comment by psteckler** [comment]
This change is for the `hex_bytes` part of the `public_key` field of `/construction/derive` and `/construction/combine`, correct?

The length is the number of hex characters, or the number of bytes they encode (2 hex chars encode 1 byte)?

**Comment by bkase** [comment]
Yes 32 bytes encoded in hex characters (maybe you can hex-decode before checking the 32/64 length or first check the length as 64/128)

**Comment by psteckler** [comment]
Can we get a test input, with the hex bytes and the Base58Check representation?

**Comment by robinbb** [comment]
@psteckler I see the merged #9965. Does that mean that this issue ought to be closed, or is there more to do?
